src/main/java/org/elasticsearch/river/kafka/IndexDocumentProducer.java

```
@@ -69,10 +81,50 @@ public void addMessagesToBulkProcessor(final Set<MessageAndMetadata> messageSet)
                         break;
                     case JSON:
                         final Map<String, Object> messageMap = reader.readValue(messageBytes);
+                        String id = null;
+                        try {
```

Do we need the exception handling? What happens if you let the exception propagate up? Usually it's better to only have exception handling at the top of the call stack. Ideally River would do something sensible like logging the exception and dropping the message, and then we wouldn't have to do explicit exception handling.